src/GLua/AG/PrettyPrint.ag

```diff
@@ -519,18 +539,43 @@ sem Expr
                         loc.prettySingle = braces @lhs.ppconf @loc.emptyContents @fields.pretty
                         loc.emptyContents = toEmpty $ null @fields.copy
                         fields.indent = @lhs.indent + (if @fields.isMultiline then 1 else 0)
+                        fields.neighborPrecedence = 0
     | BinOpExpr         lhs.pretty = @left.pretty <-> @op.pretty <-> @right.pretty
+                        loc.precedence = case @op.copy of
+                                            AOr -> 1
```

Comment:
Instead of a case statement, this could be a 

```
sem BinOp
  | AOr lhs.precedence = 1
  | AAnd lhs.precedence = 2
```

etc.

I can't blame you for not knowing this, though. Attribute grammars are a hella obscure niche thing.

Reply:
Yeah I never even wrote haskell before this lol